util/tr_openstudio.rb

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/ruby
+
+# Simple OSMeasure Test Runner
+# 2016.10.06 RPG/NREL
+# TODO add a scanner(s) for Rspec et al. test defs
+
+require 'optparse'
+require 'csv'
+require 'json'
+
+# options parser
+
+options = {}
+
+optparse = OptionParser.new do|opts|
+
+  opts.banner = "\nUsage: #{$0} [options]\n\nScans all specified test files for (Minitest) tests and \
+  \nruns them; errors reported to file."
+   
+  opts.separator ""
+  opts.separator "Options:"
+  
+  options[:wd] = Dir.pwd
+  opts.on( '-d', '--directory <value>', "Working directory (default: '.')") do|f|
+    options[:wd] = f
+  end
+  
+  options[:test_file] = nil
+  opts.on( '-t', '--test <file1,file2...>', "Test filename(s)" ) do|f|
+    options[:test_file] = f.split(",")
+  end
+
+  options[:tests_map] = nil
+  opts.on( '-m', '--mask <regex>', "Regex search for test file(s)" ) do|f|
+    options[:tests_map] = f
+  end
+
+  options[:envs] = ["lib"]
+  opts.on( '-e', '--env <path1,path2,...>', "Ruby environment path(s) (default: 'lib')" ) do|f|
+    options[:envs] = f.split(",")
+  end
+  
+  opts.on( '-f', '--fetch', "Fetch measures from BCL and exit" ) do|f|
+    options[:bcl_fetch] = f
+  end
+
+  options[:bcl_query] = 'NREL'
+  opts.on( '-b', '--bcl <query>', "BCL measure query (default: 'NREL')" ) do|f|
+    options[:bcl_query] = f
+  end
+
+  options[:test_bcl] = false
+  opts.on( '-a', '--auto', "Test NREL BCL measures" ) do|f|
```

Comment:
If you _just_ want to _get_ the measures, you run '-f'. If you want to run the NREL tests in the bcl, you run '-a', which fetches the measures **and** runs the tests. Basically '-a' is there for a bot, or for anyone to simply get all NREL measures and run their tests.